packages/web-react/src/hooks/useClickOutside.ts

```diff
@@ -3,16 +3,27 @@ import { useEffect, useCallback, MutableRefObject } from 'react';
 export interface UseClickOutsideProps {
   ref: MutableRefObject<HTMLElement | null>;
   callback?: (event: Event) => void;
+  checkClickableElement?: boolean;
 }
 
-export const useClickOutside = ({ ref, callback }: UseClickOutsideProps): void => {
+export const useClickOutside = ({ ref, callback, checkClickableElement = true }: UseClickOutsideProps): void => {
+  const isTargetClickable = (target: HTMLElement) => {
+    const tagName = target.tagName.toLowerCase();
+
+    return tagName === 'button' || tagName === 'a';
```

Comment:
I'm afraid we are missing the target here. When I click **inside the stacked Modal** anywhere except the active elements, it's the **underlying Modal what gets closed**, not the stacked one:

https://github.com/lmc-eu/spirit-design-system/assets/5614085/ae29d9cb-e67f-46e6-9185-473c92c73357

This makes me think we have to rather focus on event bubbling, not the nature of the elements being clicked on.

Reply:
This is very strange because this is exactly what I tested. 

So I'm looking for a solution